### PR TITLE
HDDS-8014. [Snapshot] Implement Native Acl Integration tests for Snapshotting

### PR DESCRIPTION
**Pull request by mladjan-gadzic**
## What changes were proposed in this pull request?

Added native ACL integration tests for Snapshotting.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8014

## How was this patch tested?

Integration tests.

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneManagerSnapshotAcl.java [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+  @ParameterizedTest
+  @EnumSource(BucketLayout.class)
+  public void testLookupKeyWithNotAllowedUser(BucketLayout bucketLayout)
+      throws Exception {
+    // GIVEN
+    setup(bucketLayout);
+    final OmKeyArgs snapshotKeyArgs = getOmKeyArgs(true);
+    final OmKeyArgs keyArgs = getOmKeyArgs(false);
+
+    // WHEN
```
Comment (GeorgeJahad) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneManagerSnapshotAcl.java:199]: Comment should be something like:  "when reading from snapshot, read disallowed."

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneManagerSnapshotAcl.java [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+    UserGroupInformation.setLoginUser(UGI2);
+    final OMException ex = Assertions.assertThrows(OMException.class,
+        () -> ozoneManager.lookupKey(snapshotKeyArgs));
+
+    // THEN
+    Assertions.assertEquals(OMException.ResultCodes.PERMISSION_DENIED,
+        ex.getResult());
+    Assertions.assertDoesNotThrow(() -> ozoneManager.lookupKey(keyArgs));
```
Comment (GeorgeJahad) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneManagerSnapshotAcl.java:207]: add a comment: "when same user reads  same key from active fs, read allowed."

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneManagerSnapshotAcl.java [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+    // THEN
+    Assertions.assertEquals(OMException.ResultCodes.PERMISSION_DENIED,
+        ex.getResult());
+    Assertions.assertDoesNotThrow(
+        () -> ozoneManager.getKeyInfo(omKeyArgs, assumeS3Context));
+  }
+
+  private static Stream<Arguments> getListStatusArguments() {
+    return Stream.of(
+        arguments(BucketLayout.DEFAULT, false, false),
```
Comment (GeorgeJahad) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneManagerSnapshotAcl.java:293]: The default bucketlayout can change, so make these: OBJECT_STORE

**Comment by GeorgeJahad** [comment]
After fixing the minor nits above, this looks good to me.

**Comment by GeorgeJahad** [comment]
LGTM

**Comment by aswinshakil** [comment]
Thanks for the patch @mladjan-gadzic, The changes look good to me. One suggestion is to move all the helper methods down, right now it looks scattered.

**Review by aswinshakil — COMMENTED** [review]
Thanks for the patch @mladjan-gadzic, The changes look good to me. One suggestion is to move all the helper methods down, right now it looks scattered.

**Comment by GeorgeJahad** [comment]
> One suggestion is to move all the helper methods down, right now it looks scattered.

+1

**Comment by mladjan-gadzic** [comment]
> Thanks for the patch @mladjan-gadzic, The changes look good to me. One suggestion is to move all the helper methods down, right now it looks scattered.

@GeorgeJahad @aswinshakil thank you for the review! I will group helper methods toghether.

**Review by prashantpogde — APPROVED** [review]
thank you @mladjan-gadzic  for adding new test cases. The changes look good to me once requested changes are made.

**Comment by GeorgeJahad** [comment]
+1

**Review by aswinshakil — APPROVED** [review]
LGTM+1

**Comment by mladjan-gadzic** [comment]
Thank you @prashantpogde for the review

**Pull request merged**